components/DataCard.tsx

```diff
@@ -1,37 +1,273 @@
+        );
+        let gotchiverseStats30dResponse = useSWR(
+            "/api/gotchiverse/stats/30",
+            Fetcher
+        );
+
+        let gotchivereStatsResponse = useSWR(
+            "/api/gotchiverse/stats", 
+            Fetcher
+        );
+
+        if (gotchivereStatsResponse.data && gotchiverseStats30dResponse.data && gotchiverseStats1dResponse.data && gotchiverseStats7dResponse.data) {
+            console.log(gotchivereStatsResponse.data, "collected");
+            dataArray = [gotchiverseStats1dResponse.data[dataField], gotchiverseStats7dResponse.data[dataField], gotchiverseStats30dResponse.data[dataField], gotchivereStatsResponse.data[dataField]];
+            dataArray.forEach((element, index) => {
+                dataArray[index] = element.toString();
+            })
+        }
+
+    }
+
+    retrieveData();
+
+    useEffect(() => {
+        function calculateAndSetTrend() {
+            let className : string = "body_trend_";
+            // const amountTotal : number = parseInt(dataArray[3], 10);
+            // now - the amount at that time period ago 
+            const changes : number = (parseInt(dataArray[3], 10) - parseInt(dataArray[timeLine], 10)) / parseInt(dataArray[timeLine], 10)
+            setTrend(changes * 100);
+            if (changes >= 0) {
+                className = className + "positive";
+            } else {
+                className = className + "negative";
+            };
+
+            setTrendClass(className);
+
+        }
+
+        if (dataArray.length > 0) {
+            calculateAndSetTrend();
+        }
+
+    }, [timeLine]);
+
+
+    if (dataArray.length === 0) {
         return (
             <div className = "waitingForConnection">
                 Loading...
             </div>
         )
     } else {
         return (
-            <div>
-                {data}
-            </div>
+            <section className = "wrapper">
+                <div className = "header">
+                    <div className = "data_title">{title}</div>
+                    <div className = "trend_wrapper">
+                    <Image
+                    src={ trend >= 0 ? `/static/images/trending-up.png` :`/static/images/trending-downWithoutBorder.png`}
+                    alt="trending"
+                    width="62"
+                    height="55"
+                    />
+                    </div>
+                </div>
+                <div className = "body">
+                    <div className = "body_data">{dataArray[timeLine]}</div>
+                    <div className = "body_changeOverTime">
+                        <div className = {trendClass}>
+                            +{trend.toFixed(2)}%
+                        </div>
+                    </div>
+
+                </div>
+                <div className = "footer">
+                    <div className = "buttons">
+
+                    
+                        <button className = "footer_button" onClick = {() => {
```

Comment:
avoid code duplication: create array with interval [1,7,30] and render button through array.map(e => (<Button..> </Button>))